test/lightning/projects/merge_projects_test.exs

```diff
@@ -2311,4 +2340,431 @@ defmodule Lightning.Projects.MergeProjectsTest do
+      target_project = insert(:project)
+
+      target_workflow =
+        insert(:workflow, project: target_project, name: "Workflow 1")
+
+      insert(:workflow_version,
+        workflow: target_workflow,
+        hash: "abc123",
+        source: "app"
+      )
+
+      source_project = insert(:project, parent_id: target_project.id)
+
+      refute MergeProjects.has_diverged?(source_project, target_project)
+    end
+
+    test "returns false when sandbox has empty workflows list" do
+      target_project = insert(:project)
+      source_project = insert(:project, parent_id: target_project.id)
+
+      refute MergeProjects.has_diverged?(source_project, target_project)
+    end
+
+    test "returns false when workflows have no versions" do
+      target_project = insert(:project)
+      source_project = insert(:project, parent_id: target_project.id)
+
+      insert(:workflow, project: target_project, name: "Workflow 1")
+      insert(:workflow, project: source_project, name: "Workflow 1")
+
+      refute MergeProjects.has_diverged?(source_project, target_project)
+    end
+  end
+
+  describe "merge_workflow/2 - edge case with multiple candidates and body matching" do
+    test "uses body matching when multiple structural candidates exist" do
+      source_trigger = build(:trigger, type: :webhook)
+
+      source_job_a =
+        build(:job, name: "job_a", body: "console.log('specific code');")
+
+      source_job_b =
+        build(:job, name: "job_b", body: "console.log('different code');")
+
+      source =
+        build(:workflow)
+        |> with_trigger(source_trigger)
+        |> with_job(source_job_a)
+        |> with_job(source_job_b)
+        |> with_edge({source_trigger, source_job_a})
+        |> with_edge({source_trigger, source_job_b})
+        |> insert()
+
+      target_trigger = build(:trigger, type: :webhook)
+
+      target_job_x =
+        build(:job, name: "job_x", body: "console.log('other code');")
+
+      target_job_y =
+        build(:job, name: "job_y", body: "console.log('specific code');")
+
+      target =
+        build(:workflow)
+        |> with_trigger(target_trigger)
+        |> with_job(target_job_x)
+        |> with_job(target_job_y)
+        |> with_edge({target_trigger, target_job_x})
+        |> with_edge({target_trigger, target_job_y})
+        |> insert()
+
+      result = MergeProjects.merge_workflow(source, target)
+
+      result_job_a = Enum.find(result["jobs"], &(&1["name"] == "job_a"))
+      assert result_job_a["id"] == target_job_y.id
+
+      result_job_b = Enum.find(result["jobs"], &(&1["name"] == "job_b"))
+      assert result_job_b["id"] == target_job_x.id
+    end
+
+    test "handles max iterations guard in structural matching" do
+      source_trigger = build(:trigger, type: :webhook)
+
+      source_jobs =
+        for i <- 1..10 do
+          build(:job, name: "job_#{i}", body: "fn(state => state)")
+        end
+
```

Comment:
@elias-ba I may be misunderstanding the purpose of this test, but if I add 10 to max_iterations here: https://github.com/OpenFn/lightning/blob/3747-sandbox-merge-divergence-warning/lib/lightning/projects/merge_projects.ex#L198  - no tests fail, not even this one.